This assert is well known to fail with aspell.
See also https://bugzilla.gnome.org/show_bug.cgi?id=772406

--- gspell-1.14.0/tests/unit-tests/test-checker.c.orig
+++ gspell-1.14.0/tests/unit-tests/test-checker.c
@@ -93,7 +93,6 @@
 	 */
 	correctly_spelled = gspell_checker_check_word (checker, "spell-checking", -1, &error);
 	g_assert_no_error (error);
-	g_assert_true (correctly_spelled);
 
 	correctly_spelled = gspell_checker_check_word (checker, "nrst-auie", -1, &error);
 	g_assert_no_error (error);
